packages/circuit-ui/styles/style-mixins.ts

```diff
@@ -196,8 +202,29 @@ export const hideVisually = (): SerializedStyles => css`
 /**
  * Visually communicates to the user that an element is focused.
  */
-export const focusOutline = (args: ThemeArgs): SerializedStyles => {
-  const theme = getTheme(args);
+
+export function focusOutline(
+  options: 'inset',
+): (args: ThemeArgs) => SerializedStyles;
+export function focusOutline(args: ThemeArgs): SerializedStyles;
+export function focusOutline(
```

Comment:
What's happening here with the three `focusOutline` declarations?

Reply:
Those are [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads), a TypeScript feature. They tell TypeScript which return type to expect based on the arguments that are passed to the function.

Reply:
Oh, thanks for the nice evening reading material 😄